### PR TITLE
#163294622 signup endpoint_v2

### PR DESCRIPTION
**Pull request by jackiegitari1234**
**What does this PR do?**
- Creates an end point for a user to sign-up

**Description of Task to be completed?**
- Create an endpoint that enables a user to create an account at Questioner
- all tests created should pass

**How should this be manually tested?**
- clone the repository to your local machine
- run the commands
-  `virtualenv venv` -  to create a virtual environment
 -   `source venv/bin/activate`  to activate a virtual environment
- `pip install -r requirements.txt` to install all the application requirements
 - `flask run`
-  use postman to test the endpoint

What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2235314
#163054752

*Screenshot*
rgstr

### CODE REVIEW

**Comment by SolomonMacharia** [comment]
looks nice. You may consider generating hashed passwords for better security.

**Comment by jackiegitari1234** [comment]
> looks nice. You may consider generating hashed passwords for better security.

thanks

**Comment by wangonya** [comment]
Looks good 👍🏾

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - "3.6"
+install:
+  - pip install pytest
+  - pip install pytest-cov
+  - pip install coveralls
+  - pip install -r requirements.txt
+script:
+  - pytest --cov=app/api
+after_success:
+  - coveralls
```
Comment (RuthNjeri) [inline .travis.yml]: add an empty line after line 12

File: app/api/v2/models/Basemodel.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#         execCommand(query)
+
+#     def Update_Row(self, query):
+#         execCommand(query)
+
+#     def Read_Row_all(self, query):
+#         execCommand(query)
+
+#     def Read_Row_one(self, query):
+#         execCommand(query)
```
Comment (RuthNjeri) [inline app/api/v2/models/Basemodel.py]: If you are not using this file you should delete it. A commented code should not be pushed

**Review by RuthNjeri — CHANGES_REQUESTED** [review]

File: app/api/v2/models/auth_model.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        self.othername = args[2]
+        self.username = args[3]
+        self.email = args[4]
+        self.phone_number = args[4]
+        self.password = args[5]
+        self.db = init_db()
+
+    def register_user(self):
+        new_user = {
+            # "public_id": str(uuid.uuid4()),
```
Comment (RuthNjeri) [inline app/api/v2/models/auth_model.py]: Do not push commented code.

Comment (RuthNjeri) [inline app/api/v2/models/auth_model.py]: Do not push commented code.

File: app/api/v2/utils/database.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+config = os.getenv("FLASK_ENV")
+
+def init_db():
+    try:
+        if config == "development":
+            db_url = os.getenv("DATABASE_URL")
+        else:
+            db_url="dbname='testingdb' host='127.0.0.1' port='5433' user='postgres' password='12345'"
```
Comment (RuthNjeri) [inline app/api/v2/utils/database.py]: I think you can also create a `TEST_DATABASE_URL` just like how you have created the development `DATABASE_URL` and get it from the env.

**Comment by RuthNjeri** [comment]
In what you have listed as **How should this be manually tested?**  in the PR details:

1. `source venv/bin/activate` to create a virtual environment - Does not create a virtual environment, it activates it. 
2. You should describe how to create a virtual environment before activating it.
3. You should also specify how to install the dependencies from the `requirements.txt` into the virtual environment.

File: Instance/config.py [hunk 5]
```diff
@@ -1,18 +1,21 @@
 import os
 '''configuration options'''
 
+
 class Config(object):
     '''Parent configuration class'''
     DEBUG = False
     SECRET_KEY = 'tryandguess'
     ENV = 'development'
     SECRET = os.getenv('SECRET')
```
Comment (RuthNjeri) [inline Instance/config.py:9]: You have a `SECRET_KEY` and a `SECRET` from the environment. You should pick one, preferably one that is hidden in the env.

**Pull request merged**